src/status_im/chat/views/message_input.cljs

```diff
@@ -40,9 +42,15 @@
    :multiline              true
    :editable               (not disable?)})
 
-(defn command-input-options [{:keys [icon-width disable? sending-disabled?]}]
+(defn command-input-options [{:keys [icon-width disable? sending-disabled? flag]}]
   {:style             (st-response/command-input icon-width disable?)
-   :on-change-text    (when-not disable? command/set-input-message)
+   :on-change-text    (when-not disable? (fn [text]
+                                           ;; update flag only when the first
+                                           ;; character was added to commands input
+                                           (when (and (s/starts-with? text command-prefix)
+                                                      (= 3 (count text)))
```

Comment:
`3` here looks a bit like magic constant, and since it can be calculated instead of hardcoded, it's probably better to replace it with `(inc (length status-im.commands.utils/command-prefix))`.
but it is up to you